avro-serializer/src/main/java/io/confluent/kafka/serializers/KafkaAvroDeserializerConfig.java

```diff
@@ -28,13 +28,19 @@
   public static final boolean SPECIFIC_AVRO_READER_DEFAULT = false;
   public static final String SPECIFIC_AVRO_READER_DOC =
       "If true, tries to look up the SpecificRecord class ";
+  
+  public static final String AVRO_FORCE_NEW_SPECIFIC_DATA_CONFIG = "force.new.specific.avro.instance";
+  public static final boolean AVRO_FORCE_NEW_SPECIFIC_DATA_DEFAULT = false;
+  public static final String AVRO_FORCE_NEW_SPECIFIC_DATA_DOC = "If true, it passes a new instace of SpecificData to SpecificDatumReader ";
```

Comment:
Good to see that this is at `Importance.LOW`, but the docs could probably be clearer. This is a very specific scenario where you'd need to use this setting. We should probably call out exactly when you'd want to use it.